debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/VirtualThreadsCommand.java

```diff
@@ -0,0 +1,105 @@
+import com.ibm.j9ddr.vm29.j9.DataType;
+import com.ibm.j9ddr.vm29.j9.J9ObjectFieldOffset;
+import com.ibm.j9ddr.vm29.pointer.ObjectReferencePointer;
+import com.ibm.j9ddr.vm29.pointer.PointerPointer;
+import com.ibm.j9ddr.vm29.pointer.generated.J9JavaVMPointer;
+import com.ibm.j9ddr.vm29.pointer.generated.J9ObjectPointer;
+import com.ibm.j9ddr.vm29.pointer.helper.J9ObjectHelper;
+import com.ibm.j9ddr.vm29.pointer.helper.J9RASHelper;
+import com.ibm.j9ddr.vm29.types.UDATA;
+
+/**
+ * vthreads command lists all the virtual threads.
+ *
+ * Example:
+ *     !vthreads
+ * Example output:
+ *     !continuationstack 0x00007fe78c0f9600 !j9vmcontinuation 0x00007fe78c0f9600 !j9object 0x0000000706401588 (Continuation) !j9object 0x0000000706400FB0 (VThread) - name1
+ *     !continuationstack 0x00007fe78c23aa80 !j9vmcontinuation 0x00007fe78c23aa80 !j9object 0x0000000706424F90 (Continuation) !j9object 0x0000000706424EF0 (VThread) - name2
+ *     !continuationstack 0x00007fe78c244ac0 !j9vmcontinuation 0x00007fe78c244ac0 !j9object 0x00000007064250D8 (Continuation) !j9object 0x0000000706425038 (VThread) - name3
+ *     ...
+ */
+public class VirtualThreadsCommand extends Command {
+	public VirtualThreadsCommand() {
+		addCommand("vthreads", "", "Lists virtual threads");
+	}
+
+	@Override
+	public void run(String command, String[] args, Context context, PrintStream out) throws DDRInteractiveCommandException {
+		try {
+			J9JavaVMPointer vm = J9RASHelper.getVM(DataType.getJ9RASPointer());
+			if (JavaVersionHelper.ensureMinimumJavaVersion(19, vm, out)) {
+				displayVirtualThreads(vm, out);
+			}
+		} catch (CorruptDataException | NoSuchFieldException e) {
+			throw new DDRInteractiveCommandException(e);
+		}
+	}
+
+	/**
+	 * Prints all the live virtual threads
+	 *
+	 * @param out the PrintStream to write output to
+	 */
+	private void displayVirtualThreads(J9JavaVMPointer vm, PrintStream out) throws CorruptDataException, NoSuchFieldException {
+		PointerPointer mainVirtualThread = vm.liveVirtualThreadList();
+
+		UDATA linkNextOffset = vm.virtualThreadLinkNextOffset();
+
+		if (mainVirtualThread.notNull() && mainVirtualThread.at(0).notNull()) {
+			/*
+			 * liveVirtualThreadList is a circular doubly-linked list storing all the live virtual threads.
+			 * The root node is a dummy virtual thread marking the start and the end of the list.
+			 */
+			J9ObjectPointer root = J9ObjectPointer.cast(mainVirtualThread.at(0));
+			J9ObjectPointer node = ObjectReferencePointer.cast(root.addOffset(linkNextOffset)).at(0);
+			J9ObjectFieldOffset nameOffset = J9ObjectHelper.getFieldOffset(node, "name", "Ljava/lang/String;");
+			J9ObjectFieldOffset contOffset = J9ObjectHelper.getFieldOffset(node, "cont", "Ljdk/internal/vm/Continuation;");
+			J9ObjectPointer cont = J9ObjectHelper.getObjectField(node, contOffset);
+			J9ObjectFieldOffset vmRefOffset = J9ObjectHelper.getFieldOffset(cont, "vmRef", "J");
+
+			while (!node.eq(root)) {
```

Comment:
The output format should take the address size into account:
```
		String addressFormat = "0x%0" + (UDATA.SIZEOF * 2) + "x";
		String outputFormat = "!continuationstack " + addressFormat
				+ " !j9vmcontinuation " + addressFormat
				+ " !j9object %s (Continuation) !j9object %s (VThread) - %s";
```
The argument on line 95 would become `outputFormat`.